overlord/devicestate/devicestate.go

```diff
@@ -433,7 +435,42 @@ func (ms *modelSnapsForRemodel) canHaveUC18PinnedTrack() bool {
 		(ms.newModelSnap.SnapType == "kernel" || ms.newModelSnap.SnapType == "gadget")
 }
 
-func remodelEssentialSnapTasks(ctx context.Context, st *state.State, ms modelSnapsForRemodel, deviceCtx snapstate.DeviceContext, fromChange string) (*state.TaskSet, error) {
+type remodelOptions struct {
+	localSnapsRequired bool
+}
+
+func (ro *remodelOptions) UpdateWithDeviceContext(st *state.State, snapName string, deviceCtx snapstate.DeviceContext, fromChange string, localSi *snap.SideInfo, path string, userID int, channel string, snapStateFlags snapstate.Flags) (*state.TaskSet, error) {
```

Comment:
the argument order here seems fairly odd, it should follow more what the function themselves do, especially after the order is fixed for the new snapstate functions